code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -54,3 +62,132 @@
 
 /datum/species/moth/get_scream_sound(mob/living/carbon/user)
 	return 'sound/voice/moth/scream_moth.ogg'
+
+/datum/species/moth/on_species_gain(mob/living/carbon/human/H)
+	..()
+	cocoon_action = new()
+	cocoon_action.Grant(H)
+
+/datum/species/moth/on_species_loss(mob/living/carbon/human/H)
+	..()
+	cocoon_action.Remove(H)
+	QDEL_NULL(cocoon_action)
+
+/datum/species/moth/spec_life(mob/living/carbon/human/H)
+	if(cocoon_action)
+		cocoon_action.UpdateButtonIcon()
```

Comment:
Is this needed? spec_life makes me sad

Reply:
This is copied from the oozeling code, this is what makes the button red/normal if the action is available or not.